torch/nn/utils/parametrizations.py

```diff
@@ -67,17 +69,12 @@ def _update_vectors(self, weight_mat: torch.Tensor) -> None:
         #    GAN training: loss = D(real) - D(fake). Otherwise, engine will
         #    complain that variables needed to do backward for the first forward
         #    (i.e., the `u` and `v` vectors) are changed in the second forward.
-        if self.u is None or self.v is None:  # type: ignore[has-type]
-            # randomly initialize `u` and `v`
-            h, w = weight_mat.size()
-            self.u = F.normalize(weight_mat.new_empty(h).normal_(0, 1), dim=0, eps=self.eps)
-            self.v = F.normalize(weight_mat.new_empty(w).normal_(0, 1), dim=0, eps=self.eps)
 
-        for _ in range(self.n_power_iterations):
+        for _ in range(n_power_iterations):
             # Spectral norm of weight equals to `u^T W v`, where `u` and `v`
             # are the first left and right singular vectors.
             # This power iteration produces approximations of `u` and `v`.
-            self.u = F.normalize(torch.mv(weight_mat, self.v),
+            self.u = F.normalize(torch.mv(weight_mat, self.v),      # type: ignore[has-type]
                                  dim=0, eps=self.eps, out=self.u)   # type: ignore[has-type]
             self.v = F.normalize(torch.mv(weight_mat.t(), self.u),  # type: ignore[has-type]
```

Comment:
Were you going to make this change as well, why did you decide against it?

Reply:
You mean the one about the multiplication order to get a better memory layout?

I realised that `mv`, at least in [LAPACK](https://www.netlib.org/lapack/explore-html/d1/d54/group__double__blas__level3_gaeda3cbd99c8fb834a60a6412878226e1.html), does have a flag that tells the algorithm that the matrix is transposed so, I would assume that PyTorch handles this correctly and makes this optimisation internally. That being said, I have not looked into that just yet, I'll do that tomorrow.

Reply:
So it does not make use of this, but I plan to implement this optimisation soon https://github.com/pytorch/pytorch/issues/59785